### ISSUE TITLE
Null is offered as option for item with enum in an array

### ISSUE THREAD

**Issue by diegodlh** [json-editor/json-editor#1164]
### General information
Given an array with items of type "string". If the `enum` keyword is used to define a list of possible array values, the JSON editor offers an additional `null` option to choose from.

Something similar occurs if the `enum` keyword is used for a property of type `object`. However, in this case, the `null` option may be removed by defining the property as required.

### Expected behavior
If a list of possible values is defined using the `enum` keyword, choosing `null` should not be an option.

### Actual behavior
Null is offered, in some cases.

### Steps to reproduce the behavior
Direct link to example: [link](https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUMDGCA2MBGqIAZglAIYDuApomALZUCsIANOHgFZUZQD62ZAJ5gArlELwwAJzplsrEIgwALKrNSgAJEtXqUIZVCgREKAPRmOteAFodasgDppAczMATKWSJQzAJgAGQJsARj8ze3U2KEEIKkIwTm5xNggpSCopWBoNEEl4EWx5NBAYuISkngU0jKyYHJKaiFyy+P1EKCkYeBcFKgK6VABtcHh46IowBShlKSp4gF0AXyW2OYBHERg592GQJpBltgKikJbYtvYuKtT0uLqG0AOS1sIOrp6+gb2EcdLJ6azeaHFarPKFbB+c7lfRkKReQQKOjdACSUDUyBQITYMHRdExoFe7U63V6bH6IkGKBGv2mAOiQMWoLBiGUYAovEy6SkmJA3XRXh4cHggLUlzwYDAUHeZAgABYQCsgA) 

```javascript
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "type": "object",
  "properties": {
    "nonull": {
      "type": "object",
      "properties": {
        "prop": {
          "type": "string",
          "enum": [
            "one",
            "two",
            "three"
          ]
        }
      },
      "required": ["prop"]
    },
    "null1": {
      "type": "object",
      "properties": {
        "prop": {
          "type": "string",
          "enum": [
            "one",
            "two",
            "three"
          ]
        }
      }
    },
    "null2": {
      "type": "array",
      "minItems": 1,
      "items": {
        "type": "string",
        "enum": [
          "one",
          "two",
          "three"
        ]
      }
    }
  }
}
```

**Comment by germanbisurgi** [comment]
That's right. I have investigated this bug and the problem extends to radio editors as well. No additional values should be allowed or inserted into selects or radios if not  specified in enums constrains.

**Comment by germanbisurgi** [comment]
Fixed in #1269

**Comment by jrkhan** [comment]
I believe this was actually a breaking change for some use cases.
We have schema that use an array of enum options (non-required) - which prior to this change would allow users to select the empty value (indicating the non required field does in fact not have a value).

This seems like a reasonable interpretation (one of several) for how to handle a non required enum values. For future readers, if you had non-required enums that were non-required - after this change ((believe 2.9.1 onwards) it's no longer possible to select an empty value.

Might be better addressed in separate issue, but I'd be interested in discussing if a flag might make sense, an in our case updating our schema to have an explicit empty value might not be viable.
